### PR TITLE
fix: add opened property change listener for all Notification constructors

### LINKED ISSUE #5418: Notification addOpenedChangeListener not working
### Description

When using a Notification and adding a OpenedChangeListener the Event will not be handled.

### Expected outcome

The added Listener should run the method defined in the lambda.

### Minimal reproducible example

```java
Button button = new Button("Try it");
button.addClickListener(clickEvent -> {
    button.setEnabled(false);
    Notification notification = new Notification("Done", 1000, Notification.Position.MIDDLE);
    notification.setPosition(Notification.Position.MIDDLE);
    notification.addDetachListener(detachEvent -> button.setEnabled(true));
    notification.addOpenedChangeListener(event -> {
        System.out.println(event.isOpened()? "notifcation was opened" : "notifcation was closed");
    });

    notification.open();
});
add(button)
```

### Steps to reproduce

Add the snippet above in a view and check the console

### Environment

Vaadin version(s): 24.2.0.alpha9
OS: Windows 11

### Browsers

Issue is not browser related

### PR DESCRIPTION
**Pull request by sujoykd**
## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes #5418

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.
- [ ] I have not completed some of the steps above and my pull request can be closed immediately.

### CODE REVIEW

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 1]
```diff
@@ -64,6 +64,8 @@ public class Notification extends Component implements HasComponents, HasStyle,
 
     private static final int DEFAULT_DURATION = 5000;
     private static final Position DEFAULT_POSITION = Position.BOTTOM_START;
+    private static final String OPENED_PROPERTY = "opened";
+    private static final String OPENED_CHANGED_PROPERTY = "opened-changed";
```
Comment (web-padawan) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:68]: This is not a property but event name, consider using `OPENED_CHANGED_EVENT`:

```suggestion
    private static final String OPENED_CHANGED_EVENT = "opened-changed";
```

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 2]
```diff
@@ -426,7 +430,7 @@ && getElement().getNode().getParent() == null) {
                         }
                     });
         }
-        getElement().setProperty("opened", opened);
+        getElement().setProperty(OPENED_PROPERTY, opened);
```
Comment (web-padawan) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:433]: Personally I don't see a need to replace `"opened"` with this constant. 
In other components we are using `getElement().getProperty("opened")`.

Comment (sujoykd) [reply vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:433]: Agreed, however sonar complains about duplicate string literal - https://github.com/vaadin/flow-components/runs/16713099188

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java (hunk 1 above)
Comment (sujoykd) [reply vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java]: done

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java (hunk 2 above)
Comment (web-padawan) [reply vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:433]: OK, let's keep it this way then.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**